discord/ext/commands/core.py

```diff
@@ -509,11 +509,18 @@ def _get_converter(self, param):
 
     async def _resolve_default(self, ctx, param):
         try:
-            if inspect.isclass(param.default) and issubclass(param.default, defaults.CustomDefault):
-                instance = param.default()
-                return await instance.default(ctx=ctx, param=param)
-            elif isinstance(param.default, defaults.CustomDefault):
-                return await param.default.default(ctx=ctx, param=param)
+            if isinstance(param.default, list) and len(param.default) > 0:
```

Comment:
This would break users who (however inadvisable it is) set their default to a list, e.g.

```py
@command
async def my_thing(ctx, argument = [1, 2, 3]):
  for  item in argument:
    do_stuff(item)
```

While the shorthand ` = [SomeDefault()]` seems neat, imo it also brings the expectation e.g., should I be able to have two or more items in that list, e.g. ` = [ RandomUser(), RandomUser() ]` for two random users by default, etc. This is, IMO, a separate feature from just fixing greedy, which should only need the below, so it should also be split out into a different commit.

Reply:
e.g., without this extra code, one could already make a default converter that just returns a list

```py
@command
async def my_think(ctx, member: Greedy[Member] = Call(lambda ctx: [ctx.author]):
  pass
```

Or 

```py
@command
async def my_think(ctx, member: Greedy[Member] = ListDefault(Author, RandomMember)):
 pass
```

Reply:
Well, I thought of consistency here, because the Greedy returns as a list, so I thought that it'll be better if it's also a list, reducing a check that's needed to check if the passed param is a list or a normal object.

Reply:
Yeah but like, e.g., there's nothing currently preventing you from setting a greedy default to a non-list. Or does it do a list conversion?

Reply:
Well, you have to do like this.
```
def command(ctx, member: Greedy[Member] = Author):
    if member is list:
        ctx.send(f"{member[0].name}")
    else:
        ctx.send(f"{member[0].name}")
```
And if they change the default like this.
```
def command(ctx, member: Greedy[Member] = [Author]):  # Still they have to do the check because we don't pass it in list. Do we?
    if member is list:
        ctx.send(f"{member[0].name}")
    else:
        ctx.send(f"{member[0].name}")
```